src/asynchronous/client.rs

```diff
@@ -74,8 +74,9 @@ impl Client {
 
         let (tx, mut rx): (ResultSender, ResultReceiver) = mpsc::channel(100);
 
-        // TODO: check return.
-        self.streams.lock().unwrap().insert(stream_id, tx);
+        self.streams.lock().map(|mut x| {
```

Comment:
How about using question mark instead of map(), I think it is more readable.

        self.streams
            .lock()
            .map_err(|_| Error::Others("Failed to acquire lock on streams".to_string()))?
            .insert(stream_id, tx);